### ISSUE TITLE
valueerror:too many file descriptors in select()

### ISSUE THREAD

**Issue by hk8805** [websocket-client/websocket-client#963]
Hello,
<img width="849" alt="image" src="https://github.com/websocket-client/websocket-client/assets/24692695/2425efc4-a69f-4cc3-b854-ca8d52db78b3">

 When i run the code above,i get an error after one day.
<img width="994" alt="image" src="https://github.com/websocket-client/websocket-client/assets/24692695/2d285eec-29d0-4845-872b-428779d666a5">
Looking for assistance.
Thank you.

**Comment by bubbleboy14** [comment]
Hey @hk8805 try it without the "rel.signal" line - that's already being taken care of here: https://github.com/websocket-client/websocket-client/blob/69e470fee083cd4d5f42bccaff3a6168bfeee863/websocket/_app.py#L142

Anyway, give that a shot and LMK how it goes!

**Comment by bubbleboy14** [comment]
Hello again @hk8805! I just reproduced this in a Windows 10 virtualbox, and fixed it in a new rel release, [0.4.9.3](https://pypi.org/project/rel/).

So upgrade to the latest rel:

` pip install rel -U `

And give it another shot, should be fixed - LMK how it goes!

**Comment by hk8805** [comment]
> Hello again @hk8805! I just reproduced this in a Windows 10 virtualbox, and fixed it in a new rel release, [0.4.9.3](https://pypi.org/project/rel/).
> 
> So upgrade to the latest rel:
> 
> `pip install rel -U`
> 
> And give it another shot, should be fixed - LMK how it goes!

you mean i try both two solution at the same time?

**Comment by hk8805** [comment]
> Hello again @hk8805! I just reproduced this in a Windows 10 virtualbox, and fixed it in a new rel release, [0.4.9.3](https://pypi.org/project/rel/).
> 
> So upgrade to the latest rel:
> 
> `pip install rel -U`
> 
> And give it another shot, should be fixed - LMK how it goes!

thank you for your reply.
you mean i try both two solution at the same time?

**Comment by bubbleboy14** [comment]
Yes that's right - you can remove the "rel.signal" line from your code (as that is already accounted for by websocket-client).

And also please update rel to the latest version - this will essentially prevent the "too many file descriptors" by batching those select()s.

Good luck, LMK how it goes!
